diff --git a/packages/apps/Settings/src/com/android/settings/DeviceInfoSettings.java b/packages/apps/Settings/src/com/android/settings/DeviceInfoSettings.java
index e9b0959..7256d9e 100644
--- a/packages/apps/Settings/src/com/android/settings/DeviceInfoSettings.java
+++ b/packages/apps/Settings/src/com/android/settings/DeviceInfoSettings.java
@@ -88,7 +88,7 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment {
         setValueSummary(KEY_EQUIPMENT_ID, PROPERTY_EQUIPMENT_ID);
         setStringSummary(KEY_DEVICE_MODEL, Build.MODEL);
         setStringSummary(KEY_BUILD_NUMBER, Build.DISPLAY);
-        findPreference(KEY_KERNEL_VERSION).setSummary(getFormattedKernelVersion());
+        findPreference(KEY_KERNEL_VERSION).setSummary(getKernelVersion());
         setValueSummary(KEY_CYANOGENMOD_VERSION, "ro.cm.version");
         setValueSummary(KEY_PARANOID_VERSION, "ro.pa.version");
         findPreference(KEY_PARANOID_VERSION).setEnabled(true);
@@ -235,6 +235,20 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment {
         }
     }
 
+    private String getKernelVersion() {
+        String procVersionStr;
+        try {
+            procVersionStr = readLine(FILENAME_PROC_VERSION);
+            return procVersionStr;
+        } catch (IOException e) {
+            Log.e(LOG_TAG,
+                "IO Exception when getting kernel version for Device Info screen",
+                e);
+
+            return "Unavailable";
+        }
+    }
+
     private String getFormattedKernelVersion() {
         String procVersionStr;
 
